packages/google_maps_flutter/ios/Classes/GoogleMapMarkerController.m

```diff
@@ -61,3 +70,172 @@ - (void)setZIndex:(int)zIndex {
   _marker.zIndex = zIndex;
 }
 @end
+
+static void interpretMarkerOptions(id json, id<FLTGoogleMapMarkerOptionsSink> sink,
+                                   NSObject<FlutterPluginRegistrar>* registrar) {
+  NSDictionary* data = json;
+  id alpha = data[@"alpha"];
+  if (alpha) {
+    [sink setAlpha:toFloat(alpha)];
+  }
+  id anchor = data[@"anchor"];
+  if (anchor) {
+    [sink setAnchor:toPoint(anchor)];
+  }
+  id draggable = data[@"draggable"];
+  if (draggable) {
+    [sink setDraggable:toBool(draggable)];
+  }
+  id icon = data[@"icon"];
+  if (icon) {
+    NSArray* iconData = icon;
+    UIImage* image;
+    if ([iconData[0] isEqualToString:@"defaultMarker"]) {
```

Comment:
it would crash if iconData is size 0. If you only want to escape this if block, then maybe use .firstObject of NSArray which returns nil when the array is nil or empty; and makes the if statement return NO(false).
```suggestion
    if ([iconData.firstObject isEqualToString:@"defaultMarker"]) {
```